src/order/rel_classes.lean

```diff
@@ -311,3 +281,173 @@ instance is_trans_preimage_snd {r : α → α → Prop} [h : is_trans α r] :
   is_trans (α × α) (prod.snd ⁻¹'o r) := ⟨λ _ _ _, trans_of r⟩
 
 end prod
+
+/-! ### Strict-non strict relations -/
+
+/-- An unbundled relation class stating that `r` is the nonstrict relation corresponding to the
+strict relation `s`. Compare `preorder.lt_iff_le_not_le`. This is mostly meant to provide dot
+notation on `(⊆)` and `(⊂)`. -/
+class is_nonstrict_strict_order (α : Type*) (r s : α → α → Prop) :=
+(right_iff_left_not_left (a b : α) : s a b ↔ r a b ∧ ¬ r b a)
+
+lemma right_iff_left_not_left {r s : α → α → Prop} [is_nonstrict_strict_order α r s] {a b : α} :
+  s a b ↔ r a b ∧ ¬ r b a :=
+is_nonstrict_strict_order.right_iff_left_not_left _ _
+
+/-- A version of `right_iff_left_not_left` with explicit `r` and `s`. -/
+lemma right_iff_left_not_left_of (r s : α → α → Prop) [is_nonstrict_strict_order α r s] {a b : α} :
+  s a b ↔ r a b ∧ ¬ r b a :=
+right_iff_left_not_left
+
+-- The free parameter `r`  is `out_param`s so this is not dangerous.
+@[priority 100, nolint dangerous_instance] -- see Note [lower instance priority]
+instance is_nonstrict_strict_order.to_is_irrefl {r : out_param $ α → α → Prop} {s : α → α → Prop}
```

Comment:
Putting `out_param` on an instance does nothing, the elaborator only checks for `out_param` on the class before starting to consider instances. So we might put it on 
```suggestion
-- This has a free parameter `r` which is fine because it can be uniquely inferred from `s`.
@[priority 100, nolint dangerous_instance] -- see Note [lower instance priority]
instance is_nonstrict_strict_order.to_is_irrefl {r : α → α → Prop} {s : α → α → Prop}
```

Reply:
Same here. `r` can be uniquely inferred from `s`, although it should never come up in practice.